boards: ti: lp_am243x: initial support

spencerjcoward opened the pull request:
Add support for the TI AM243x LaunchPad board. This board is based on the work done for the am243x_evm board, with modifications to the device tree and board configuration files to suit the LaunchPad's hardware.

natto1784 commented on boards/ti/lp_am243x/lp_am243x_am2434_r5f0_0-pinctrl.dtsi:
```diff
@@ -0,0 +1,35 @@
+	};
+};
+
+&pinctrl {
```
same node as above

spencerjcoward replied:
agree, fixed here eb68a988af2094473948e76e466087f4e627f43d

m-braunschweig reviewed (changes requested):
Seems mosty good except some old EVM references. I have problems running the blinky sample with additionally `CONFIG_LOG` and `CONFIG_ASSERT` enabled. Since the Data Fault Address Register held an address in the external RAM memory space I assume that something like RSC tried to log data and this resulted in a fault.

Aditionally I think it might be better to name the SoC before the fact that it's a launchpad so it's consistent with the 243evm.

Also as Natto mentioned I also worked on adding Zephyr support for this board (that's also intended to be added upstream). However I didn't take a look at the M4 processor core and ADC/I2C so I think it would be better to continue with this PR

spencerjcoward committed:
boards: ti: lp_am243x: initial support

Add support for the TI AM243x LaunchPad board. This board is based on the
work done for the am243x_evm board, with modifications to the device tree
and board configuration files to suit the LaunchPad's hardware. With
reduced image size for board.

Signed-off-by: Spencer J. Coward <spencerjcoward@gmail.com>

spencerjcoward commented on boards/ti/lp_am243x/doc/index.rst:
```diff
@@ -0,0 +1,275 @@
+- 800MHz for R5F0_0
+- 400MHz for M4
+
+DDR RAM
+-------
+
+The board has 2GB of DDR RAM available. This board configuration allocates:
+
+- 4KB Resource Table at 0xa4100000 for M4
+- 4KB Resource Table at 0xa0100000 for R5F0_0
+- 8MB Shared Memory at 0xa5000000 for inter-processor communication
```
i see that i missed these non existent  ddr locations still being referenced in the dts, how would you like me to resolve the current need for RSC_TABLE? SOC_SERIES_AM6X_M4 selects this in Kconfig, should i reserve a piece of sram to contain these?

spencerjcoward replied:
also i did do a test where i remove all the nodes referencing the external ddr (ipc, shm, rsc table) and put OPENAMP_RSC_TABLE=n in the defconfig for the m4 and then i am able to turn on CONFIG_LOG and CONFIG_ASSERT.

Would it be best to leave the RSC table and IPC for another PR and just disable that feature for now?

m-braunschweig replied:
I think for RSC / ipc it would be good to put the memory regions into one of the unused SRAM areas (based on the default layout of the TI MCU+ SDK). If you want to do it in this PR on in another PR is up to you.

> also i did do a test where i remove all the nodes referencing the external ddr (ipc, shm, rsc table) and put OPENAMP_RSC_TABLE=n in the defconfig for the m4 and then i am able to turn on CONFIG_LOG and CONFIG_ASSERT.

I will create a separate comment for this since there are multiple problems (also in already existing code). Please also note that I'm focussing on using Zephyr on the Cortex-R5 and no the M4

natto1784 replied:
An offtopic suggestion: Maybe we ought to put the C-file MPU configuration in `boards/` directory instead of `soc/`. This would help avoid `ifdef`s as well as would allow a more flexible MPU configuration. Right now, for AM2434, it is a mix of DT configuration and C-file which does not sit right with me.

spencerjcoward committed:
boards: ti: lp_am243x: dtsi and board.cmake formatting cleanup

Address comments from PR 91986, relating to the formatting of the
board.cmake file and the dtsi files for the lp_am243x board.

Signed-off-by: Spencer J. Coward <spencerjcoward@gmail.com>

spencerjcoward commented on boards/ti/lp_am243x/doc/img/lp-am243-angled.webp:
done

nordicjm commented on boards/ti/lp_am243x/board.cmake:
```diff
@@ -3,9 +3,9 @@
 
 if(CONFIG_SOC_AM2434_M4)
-board_runner_args(openocd "--no-init" "--no-halt" "--no-targets" "--gdb-client-port=3338")
+  board_runner_args(openocd "--no-init" "--no-halt" "--no-targets" "--gdb-client-port=3338")
```
no fixup commits, squash

natto1784 commented on boards/ti/lp_am243x/lp_am243x_am2434_r5f0_0.dts:
```diff
@@ -0,0 +1,172 @@
+	channel@4 {
+		reg = <4>;
+		ti,open-delay = <0>;
+		zephyr,gain = "ADC_GAIN_1";
+		zephyr,reference = "ADC_REF_INTERNAL";
+		zephyr,acquisition-time = <0>;
+		zephyr,resolution = <12>;
+		zephyr,oversampling = <4>;
+	};
+
+	channel@5 {
+		reg = <5>;
+		ti,open-delay = <0>;
+		zephyr,gain = "ADC_GAIN_1";
+		zephyr,reference = "ADC_REF_INTERNAL";
+		zephyr,acquisition-time = <0>;
+		zephyr,resolution = <12>;
+		zephyr,oversampling = <4>;
+	};
+
+	channel@6 {
+		reg = <6>;
+		ti,open-delay = <0>;
+		zephyr,gain = "ADC_GAIN_1";
+		zephyr,reference = "ADC_REF_INTERNAL";
+		zephyr,acquisition-time = <0>;
+		zephyr,resolution = <12>;
+		zephyr,oversampling = <4>;
+	};
+
+	channel@7 {
+		reg = <7>;
+		ti,open-delay = <0>;
+		zephyr,gain = "ADC_GAIN_1";
+		zephyr,reference = "ADC_REF_INTERNAL";
+		zephyr,acquisition-time = <0>;
+		zephyr,resolution = <12>;
+		zephyr,oversampling = <4>;
+	};
+};
```
These should not be here, I made a PR to change that for am243x evm as well: https://github.com/zephyrproject-rtos/zephyr/pull/92170
